### PR TITLE
Resource requests and limits for CPU and Memory

### PR DESCRIPTION
**Pull request by maris-accenture**
This change is [Reviewable](https://reviewable.io/reviews/att-comdev/openstack-helm/106)

### CODE REVIEW

**Comment by wilkers-steve** [comment]
Some of the charts in this pull request are included in the pull request submitted by @Ciello89.  Have you two discussed working on this effort together, or is this duplicate effort?

**Comment by v1k0d3n** [comment]
hey @maris-accenture, thanks for submitting these enhancements! unfortunately, someone else picked them up just before your PR. keep an eye on #103, and see if anything is left over. there are some nice comments/reviews that could possibly leave some things on the table.

**Comment by Ciello89** [comment]
Agree with @wilkers-steve, the only charts that haven't been addressed in the PR #103 are cinder, heat and nova.

**Comment by maris-accenture** [comment]
Thanks to all reviewers!

**Comment by maris-accenture** [comment]
Thanks to all reviewers!

**Comment by maris-accenture** [comment]
Reverted back 8 commits. Added resources and development flags to cinder, heat, and nova.

**Comment by maris-accenture** [comment]
@Ciello89
Resolved conflicts

File: cinder/templates/deployment-api.yaml [hunk 1]
```diff
@@ -51,6 +51,16 @@ spec:
         - name: cinder-api
           image: {{ .Values.images.api }}
           imagePullPolicy: {{ .Values.images.pull_policy }}
+          {{- if .Values.development.enabled }}
```
Comment (alanmeadows) [inline cinder/templates/deployment-api.yaml]: This contradicts the pattern in #103 which uses .Values.resources.enabled.

File: cinder/templates/deployment-api.yaml [hunk 2]
```diff
@@ -51,6 +51,16 @@ spec:
...
+            resources:
```
Comment (alanmeadows) [inline cinder/templates/deployment-api.yaml]: The indentation does not match L53

**Comment by maris-accenture** [comment]
@alanmeadows 
Hopefully correct changes have been made and conflicts resolved.
 Maybe it is reasonable to remove duplicated worker resources (lines 100 - 108) from heat values.yaml

File: nova/values.yaml [hunk 3]
```diff
@@ -209,6 +209,81 @@ endpoints:
     port:
         admin: 35357
         public: 5000
+resources:
```
Comment (alanmeadows) [inline nova/values.yaml]: This needs to be fixed -- an automerge has placed your additions directly in the middle of the endpoints: definition.

**Comment by alanmeadows** [comment]
@maris-accenture please go ahead and squash your commit history as this PR has been around for a bit with a few passes.

**Comment by maris-accenture** [comment]
@alanmeadows neutron endpints moved to endpoints:
Since I already did push, is it now possible to do squash?
Thanks

**Comment by maris-accenture** [comment]
Hello @alanmeadows, squash is done.

**Comment by maris-accenture** [comment]
Review status: 0 of 30 files reviewed at latest revision, 3 unresolved discussions.

---

*[cinder/templates/deployment-api.yaml, line 54 at r4](https://reviewable.io:443/reviews/att-comdev/openstack-helm/106#-KaDkJnU3vMvWN7RIPKf-r4-54:-KbU7OD4giMF6s1z6Ed3:b-896fix) ([raw file](https://github.com/att-comdev/openstack-helm/blob/02f8c3abad82bd4a2e9d4cc2bb764f4154e01116/cinder/templates/deployment-api.yaml#L54)):*
<details><summary><i>Previously, alanmeadows (Alan Meadows) wrote…</i></summary><blockquote>

This contradicts the pattern in #103 which uses .Values.resources.enabled.
</blockquote></details>

Done.

---

*[cinder/templates/deployment-api.yaml, line 55 at r4](https://reviewable.io:443/reviews/att-comdev/openstack-helm/106#-KaDkJnU3vMvWN7RIPKf-r4-55:-KbU7KmPtFcPJM7kTYQ4:b-896fix) ([raw file](https://github.com/att-comdev/openstack-helm/blob/02f8c3abad82bd4a2e9d4cc2bb764f4154e01116/cinder/templates/deployment-api.yaml#L55)):*
<details><summary><i>Previously, alanmeadows (Alan Meadows) wrote…</i></summary><blockquote>

The indentation does not match L53
</blockquote></details>

Done.

---

*[nova/values.yaml, line 212 at r5](https://reviewable.io:443/reviews/att-comdev/openstack-helm/106#-KaDkJnVa03cHL6X0cS3-r5-212:-KbU7Y0aiFsoJt5wJ4PG:b-896fix) ([raw file](https://github.com/att-comdev/openstack-helm/blob/9f72511784bfa6afe3f12551b98db2e333c3b256/nova/values.yaml#L212)):*
<details><summary><i>Previously, alanmeadows (Alan Meadows) wrote…</i></summary><blockquote>

This needs to be fixed -- an automerge has placed your additions directly in the middle of the endpoints: definition.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/att-comdev/openstack-helm/106)*

**Comment by jev-accenture** [comment]
Reviewed 1 of 30 files at r5, 29 of 30 files at r6.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[heat/templates/job-db-init.yaml, line 30 at r6](https://reviewable.io:443/reviews/att-comdev/openstack-helm/106#-KbUIuvMRBI0vJPXzsDA:-KbUIuvNRj2hzidYHFEh:b-lhwosc) ([raw file](https://github.com/att-comdev/openstack-helm/blob/e28f0a7de08d18aeb2ee62cdc39c679d27ce7c8f/heat/templates/job-db-init.yaml#L30)):*
> ```YAML
>               memory: {{ .Values.resources.heat_db_init.limits.memory | quote }}
>               cpu: {{ .Values.resources.heat_db_init.limits.cpu | quote }}
>          {{- end }}
> ```

wrong indention

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/att-comdev/openstack-helm/106)*

**Comment by maris-accenture** [comment]
Review status: 29 of 30 files reviewed at latest revision, 4 unresolved discussions.

---

*[heat/templates/job-db-init.yaml, line 30 at r6](https://reviewable.io:443/reviews/att-comdev/openstack-helm/106#-KbUIuvMRBI0vJPXzsDA:-KbUOQw_Z_1p2Im3VEFT:b-896fix) ([raw file](https://github.com/att-comdev/openstack-helm/blob/e28f0a7de08d18aeb2ee62cdc39c679d27ce7c8f/heat/templates/job-db-init.yaml#L30)):*
<details><summary><i>Previously, jev-accenture wrote…</i></summary><blockquote>

wrong indention
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/att-comdev/openstack-helm/106)*

**Comment by jev-accenture** [comment]
Reviewed 1 of 1 files at r7.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/att-comdev/openstack-helm/106)*

**Pull request merged**